OpenRA.Mods.Common/Traits/Detonates.cs

```diff
@@ -0,0 +1,63 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Orders;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Deploy order will kill the actor with an explosion.")]
+	class DetonatesInfo : ITraitInfo, Requires<ExplodesInfo>
+	{
+		[VoiceReference] public readonly string Voice = "Action";
+
+		public object Create(ActorInitializer init) { return new Detonates(init.Self, this); }
+	}
+
+	class Detonates : IIssueOrder, IResolveOrder, IOrderVoice
```

Comment:
Actually... this can be done using the existing `DeployToUpgrade` and `KillsSelf` traits.